CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+
+2. To propose and implement a new example or workflow, please file a new feature request issue. Describe the intended example/workflow and discuss the design and implementation with the team and community. Once the team agrees that the plan looks good, go ahead and implement it, using the code contributions guide below.
+
+3. To implement a feature or bug fix for an existing issue, please follow the code contributions guide below. If you need more context on a particular issue, please ask in a comment.
+
+As contributors and maintainers to this project, you are expected to abide by the project's code of conduct.
+
+## Code contributions
+
+### Your first issue
+
+1. Follow the guide at the bottom of this page for Setting up your build environment.
+2. Find an issue to work on. The best way is to look for the `good first issue` or `help wanted` labels.
+3. Comment on the issue stating that you are going to work on it.
+4. Create a fork of the repository and check out a branch with a name that describes your planned work. For example, `add-new-workflow-example`.
+5. Write code to address the issue or implement the feature.
+6. Add unit tests and documentation.
+7. Create your pull request. To run continuous integration (CI) tests without requesting review, open a draft pull request.
+8. Verify that CI passes all status checks. Fix if needed.
+9. Wait for other developers to review your code and update code as needed.
+10. Once reviewed and approved, a team member will merge your pull request.
+
+If you are unsure about anything, don't hesitate to comment on issues and ask for clarification!
+
+### Seasoned developers
+
+Once you have gotten your feet wet and are more comfortable with the code, you can look at the prioritized issues for our next release in our project boards.
+
+**Note:** Always look at the release board that is currently under development for issues to work on. This is where the team also focuses their efforts.
+
+Look at the unassigned issues, and find an issue to which you are comfortable contributing. Start with _Step 3_ above, commenting on the issue to let others know you are working on it. If you have any questions related to the implementation of the issue, ask them in the issue instead of the PR.
+
+## Setting up your build environment
+
+The following instructions are for developers and contributors to NVIDIA cuOpt Examples and Workflows development. These instructions are tested on Ubuntu Linux LTS releases. Use these instructions to build and contribute to the development. Other operating systems may be compatible, but are not currently tested.
+
+### General requirements
+
+- Python 3.8+
+- CUDA 11.8+
+- NVIDIA GPU with Volta architecture or better (Compute Capability >=7.0)
+- NVIDIA cuOpt service API access and credentials
+
+### Create the build environment
```

Comment:
Lets split this into 2 options, one is build and install using cuOpt source, lets point to read me in that directory

And another one is use a docker container, pull it and install base requirements there and mount notebook there, and run the jupyter notebooks so that can use it, I will help you with this one.

May be we can also point to getting started to this part so they can install.